### PR TITLE
ITKDev: Settings pattern

### PR DESCRIPTION
**Pull request by cableman**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-636

#### Description

Applying the settings pattern used in dpl_instance_loan as requested in https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/313#discussion_r1200271788

#### Screenshot of the result

N/A

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

N/A

### CODE REVIEW

**Review by sinejespersen — APPROVED** [review]

File: config/sync/dpl_fbs.settings.yml [hunk 1]
```diff
@@ -1 +1 @@
-base_url: 'https://fbs-openplatform.dbc.dk'
+baseUrl: 'https://fbs-openplatform.dbc.dk'
```
Comment (spaceo) [inline config/sync/dpl_fbs.settings.yml:1]: Please change this into snake case. As mentioned [here](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/250#discussion_r1209895164).

File: web/modules/custom/dpl_dashboard/src/Form/DashboardSettingsForm.php [hunk 2]
```diff
@@ -30,7 +58,7 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('dashboard.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_dashboard/src/Form/DashboardSettingsForm.php:61]: Please use `loadConfig()` as mentioned [here](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/250#discussion_r1209895164).

File: web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php [hunk 3]
```diff
@@ -66,78 +64,80 @@ public static function create(ContainerInterface $container, array $configuratio
...
+      'warning-icon-alt-text' => $this->t('warningIconAltText', [], ['context' => 'Dashboard']),
+      'your-profile-text' => $this->t('Your profile', [], ['context' => 'Dashboard']),
     ] + DplReactAppsController::externalApiBaseUrls();
 
     return [
-      "#theme" => "dpl_react_app",
-      "#name" => "DashBoard",
-      "#data" => $data,
+      '#theme' => 'dpl_react_app',
+      '#name' => 'DashBoard',
```
Comment (spaceo) [inline web/modules/custom/dpl_dashboard/src/Plugin/Block/DashboardBlock.php:139]: Please use snake case in theme name. That is what we have done in other [places](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/64d430729cd35f5a7315ae68abc16bec7dae9ff9/web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php#L149).

File: web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php [hunk 4]
```diff
@@ -30,7 +58,7 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('dpl_loan_list.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_loans/src/Form/LoanListSettingsForm.php:61]: Please use `loadConfig` as mentioned earlier. And change conf keys correspondingly.

File: web/modules/custom/dpl_publizon/src/Form/PublizonSettingsForm.php [hunk 5]
```diff
@@ -30,7 +58,7 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('dpl_publizon.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_publizon/src/Form/PublizonSettingsForm.php:61]: Please use loadConfigand change keys

File: web/modules/custom/dpl_recommender/src/Form/RecommenderSettingsForm.php [hunk 6]
```diff
@@ -30,13 +58,13 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('recommender.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_recommender/src/Form/RecommenderSettingsForm.php:61]: Please use loadConfigand change keys

File: web/modules/custom/dpl_reservations/src/Form/ReservationListSettingsForm.php [hunk 7]
```diff
@@ -30,7 +58,7 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('dpl_reservation_list.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_reservations/src/Form/ReservationListSettingsForm.php:61]: Please use loadConfigand change keys

File: web/modules/custom/dpl_fbs/src/Form/FbsSettingsForm.php [hunk 8]
```diff
@@ -30,7 +58,7 @@ public function getFormId(): string {
    * {@inheritdoc}
    */
   public function buildForm(array $form, FormStateInterface $form_state): array {
-    $config = $this->config('dpl_fbs.settings');
+    $config = $this->configService->getConfig();
```
Comment (spaceo) [inline web/modules/custom/dpl_fbs/src/Form/FbsSettingsForm.php:61]: Please use loadConfigand change keys

**Review by spaceo — CHANGES_REQUESTED** [review]
Great! I really appreciate the effort in standardising this!
This is however implementations that conflicts with the intention of the configuration handling:
When to use Drupal snake case and the camelCasing of the react config objetcs.
The idea was to use `loadConfig()` for Drupal snake case config and `getConfig()` when integrating configuration with the apps.
Indeed the method names are not self explanatory. Could be they should be renamed? 🤔

**Review by spaceo — CHANGES_REQUESTED** [review]
Great! I really appreciate the effort in standardising this!
This is however implementations that conflicts with the intention of the configuration handling:
When to use Drupal snake case and the camelCasing of the react config objetcs.
The idea was to use `loadConfig()` for Drupal snake case config and `getConfig()` when integrating configuration with the apps.
Indeed the method names are not self explanatory. Could be they should be renamed? 🤔

**Comment by cableman** [comment]
Nice to known... might need better documentation.

**Comment by cableman** [comment]
Reverted to using loadConfig and merged due to no more comments

**Pull request merged**